engine/src/flutter/shell/platform/windows/flutter_host_window.cc

```diff
@@ -0,0 +1,519 @@
+  // value indicates apps should use light mode.
+  DWORD light_mode;
+  DWORD light_mode_size = sizeof(light_mode);
+  LSTATUS const result =
+      RegGetValue(HKEY_CURRENT_USER, kGetPreferredBrightnessRegKey,
+                  kGetPreferredBrightnessRegValue, RRF_RT_REG_DWORD, nullptr,
+                  &light_mode, &light_mode_size);
+
+  if (result == ERROR_SUCCESS) {
+    BOOL enable_dark_mode = light_mode == 0;
+    DwmSetWindowAttribute(window, DWMWA_USE_IMMERSIVE_DARK_MODE,
+                          &enable_dark_mode, sizeof(enable_dark_mode));
+  }
+}
+
+// Inserts |content| into the window tree.
+void SetChildContent(HWND content, HWND window) {
+  SetParent(content, window);
+  RECT client_rect;
+  GetClientRect(window, &client_rect);
+  MoveWindow(content, client_rect.left, client_rect.top,
+             client_rect.right - client_rect.left,
+             client_rect.bottom - client_rect.top, true);
+}
+
+}  // namespace
+
+namespace flutter {
+
+std::unique_ptr<FlutterHostWindow> FlutterHostWindow::createRegularWindow(
+    WindowManager* window_manager,
+    FlutterWindowsEngine* engine,
+    const FlutterWindowSizing& content_size) {
+  DWORD window_style = WS_OVERLAPPEDWINDOW;
+  DWORD extended_window_style = 0;
+  std::optional<Size> smallest = std::nullopt;
+  std::optional<Size> biggest = std::nullopt;
+
+  if (content_size.has_constraints) {
+    smallest = Size(content_size.min_width, content_size.min_height);
+    if (content_size.max_width > 0 && content_size.max_height > 0) {
+      biggest = Size(content_size.max_width, content_size.max_height);
+    }
+  }
+
+  // TODO(knopp): What about windows sized to content?
+  FML_CHECK(content_size.has_size);
+
+  // Calculate the screen space window rectangle for the new window.
+  // Default positioning values (CW_USEDEFAULT) are used
+  // if the window has no owner.
+  Rect const initial_window_rect = [&]() -> Rect {
+    std::optional<Size> const window_size = GetWindowSizeForClientSize(
+        *engine->windows_proc_table(),
+        Size(content_size.width, content_size.height), smallest, biggest,
+        window_style, extended_window_style, nullptr);
+    return {{CW_USEDEFAULT, CW_USEDEFAULT},
+            window_size ? *window_size : Size{CW_USEDEFAULT, CW_USEDEFAULT}};
+  }();
+
+  // Set up the view.
+  auto view_window = std::make_unique<FlutterWindow>(
+      initial_window_rect.width(), initial_window_rect.height(),
+      engine->windows_proc_table());
+
+  std::unique_ptr<FlutterWindowsView> view =
+      engine->CreateView(std::move(view_window));
+  if (view == nullptr) {
+    FML_LOG(ERROR) << "Failed to create view";
+    return nullptr;
+  }
+
+  std::unique_ptr<FlutterWindowsViewController> view_controller =
+      std::make_unique<FlutterWindowsViewController>(nullptr, std::move(view));
+  FML_CHECK(engine->running());
+  // Must happen after engine is running.
+  view_controller->view()->SendInitialBounds();
```

Comment:
Is this necessary? `FlutterWindowsEngine::CreateView` sends the initial viewport metric too:

https://github.com/flutter/flutter/blob/85824038d07ba264fcd3074838ed1c894a263b19/engine/src/flutter/shell/platform/windows/flutter_windows_engine.cc#L541